src/js/stores/MesosLogStore.js

```diff
@@ -1,30 +1,62 @@
-import {Store} from 'mesosphere-shared-reactjs';
+import BaseStore from './BaseStore';
 
 import ActionTypes from '../constants/ActionTypes';
 import AppDispatcher from '../events/AppDispatcher';
 import Config from '../config/Config';
-import EventTypes from '../constants/EventTypes';
-import GetSetMixin from '../mixins/GetSetMixin';
+import EventTypes from '../constants/EventTypes';;
 import Item from '../structs/Item';
 import LogBuffer from '../structs/LogBuffer';
 import MesosLogActions from '../events/MesosLogActions';
 
 const MAX_FILE_SIZE = 50000;
 
-const MesosLogStore = Store.createStore({
-  storeID: 'mesosLog',
-
-  mixins: [GetSetMixin],
-
-  addChangeListener: function (eventName, callback) {
+class MesosLogStore extends BaseStore {
+
+  constructor() {
+    super(...arguments);
+
+    this.dispatcherIndex = AppDispatcher.register((payload) => {
```

Comment:
Maybe I miss how it really works. :)

I wonder how this is unregistered properly? The store has no deconstructor. This looks like a memory leak here.

A function gets registered at a 'global' object `AppDispatcher` in a constructor that could be instantiated n-times. 
I think nobody noticed yet, because MesosLogStore is only instantiated once?